apps/src/applab/designElements/photochooser.jsx

```diff
@@ -0,0 +1,226 @@
+  };
+
+  render() {
+    const element = this.props.element;
+
+    let iconColorPicker;
+    const canonicalImage = element.getAttribute('data-canonical-image-url');
+    if (ICON_PREFIX_REGEX.test(canonicalImage)) {
+      iconColorPicker = (
+        <ColorPickerPropertyRow
+          desc={'icon color'}
+          initialValue={elementUtils.rgb2hex(
+            element.getAttribute('data-icon-color') || '#000000'
+          )}
+          handleChange={this.handleIconColorChange}
+        />
+      );
+    }
+
+    return (
+      <div id="propertyRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
+          isIdRow
+        />
+        <PropertyRow
+          desc={'width (px)'}
+          isNumber
+          initialValue={parseInt(element.style.width, 10)}
+          handleChange={this.props.handleChange.bind(this, 'style-width')}
+        />
+        <PropertyRow
+          desc={'height (px)'}
+          isNumber
+          initialValue={parseInt(element.style.height, 10)}
+          handleChange={this.props.handleChange.bind(this, 'style-height')}
+        />
+        <PropertyRow
+          desc={'x position (px)'}
+          isNumber
+          initialValue={parseInt(element.style.left, 10)}
+          handleChange={this.props.handleChange.bind(this, 'left')}
+        />
+        <PropertyRow
+          desc={'y position (px)'}
+          isNumber
+          initialValue={parseInt(element.style.top, 10)}
+          handleChange={this.props.handleChange.bind(this, 'top')}
+        />
+        <ColorPickerPropertyRow
+          desc={'background color'}
+          initialValue={elementUtils.rgb2hex(element.style.backgroundColor)}
+          handleChange={this.props.handleChange.bind(this, 'backgroundColor')}
+        />
+        {iconColorPicker}
+        <BorderProperties
+          element={element}
+          handleBorderWidthChange={this.props.handleChange.bind(
+            this,
+            'borderWidth'
+          )}
+          handleBorderColorChange={this.props.handleChange.bind(
+            this,
+            'borderColor'
+          )}
+          handleBorderRadiusChange={this.props.handleChange.bind(
+            this,
+            'borderRadius'
+          )}
+        />
+        <BooleanPropertyRow
+          desc={'hidden'}
+          initialValue={$(element).hasClass('design-mode-hidden')}
+          handleChange={this.props.handleChange.bind(this, 'hidden')}
+        />
+        <ZOrderRow
+          element={this.props.element}
+          onDepthChange={this.props.onDepthChange}
+        />
+      </div>
+    );
+  }
+}
+
+class PhotoChooserEvents extends React.Component {
+  static propTypes = {
+    element: PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    onInsertEvent: PropTypes.func.isRequired
+  };
+
+  getPhotoselectEventCode() {
```

Comment:
NIT: the 's' in 'select' should be capitalized (and in insertPhotoselect)
Even more NITpicky: consider renaming `getPhotoSelectEventCode` to `getPhotoSelectedEventCode` (added 'ed') and `insertPhotoSelect` to `insertPhotoSelected`